homeassistant/components/qbittorrent/coordinator.py

```diff
@@ -31,8 +44,21 @@ def __init__(self, hass: HomeAssistant, client: Client) -> None:
             update_interval=timedelta(seconds=30),
         )
 
-    async def _async_update_data(self) -> dict[str, Any]:
+    def update(self) -> SessionStats:
+        """Get the latest data from qBittorrent instance."""
         try:
-            return await self.hass.async_add_executor_job(self.client.sync_main_data)
+            data = self.client.sync_main_data()
+            self.total_torrents = self.client.torrents(filter='all')
+            self.active_torrents = self.client.torrents(filter='active')
+            self.inactive_torrents = self.client.torrents(filter='inactive')
+            self.paused_torrents = self.client.torrents(filter='paused')
+            self.seeding_torrents = self.client.torrents(filter='seeding')
+            self.started_torrents = self.client.torrents(filter='started')
```

Comment:
Can we get this info from the `.sync_main_data`? Not sure which approach I like better, filtering ourselves or doing 8 requests for an update.

Reply:
We "could", but my mindset was it makes sense to use an existing API than to create out own filtering. I'm ok with changing this to doing that, but we of course want to swap back to this if we make this a service response in the future. Let me know if you'd for sure like this change

Reply:
I mean, my doubt is that there are 2 ways, doing 8 requests or filtering. One is putting pressure on the network and one is putting pressure on CPU cycles. There probably are people with 100+ torrents, if not more. We then fetch the same data multiple times.

There is no ideal way to solve this imo, but we should choose the best way possible, after taking in all factors.